instrumentor/controllers/deleteinstrumentationconfig/source_controller.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deleteinstrumentationconfig
+
+import (
+	"context"
+	"errors"
+
+	"github.com/odigos-io/odigos/api/odigos/v1alpha1"
+	"github.com/odigos-io/odigos/k8sutils/pkg/consts"
+	k8sutils "github.com/odigos-io/odigos/k8sutils/pkg/utils"
+	"github.com/odigos-io/odigos/k8sutils/pkg/workload"
+
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+type SourceReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+}
+
+func (r *SourceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("Reconciling Source object", "name", req.Name, "namespace", req.Namespace)
+
+	source := &v1alpha1.Source{}
+	err := r.Get(ctx, req.NamespacedName, source)
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+
+	// If this is a regular Source that's being deleted, or a workload Exclusion Source
+	// that's being created, try to uninstrument relevant workloads.
+	if source.DeletionTimestamp.IsZero() == v1alpha1.IsWorkloadExcludedSource(source) {
+		logger.Info("Reconciling workload for Source object", "name", req.Name, "namespace", req.Namespace)
+
+		if source.Spec.Workload.Kind == "Namespace" {
+			logger.V(2).Info("Uninstrumenting workloads for Namespace Source", "name", req.Name, "namespace", req.Namespace)
+
+			for _, kind := range []workload.WorkloadKind{
+				workload.WorkloadKindDaemonSet,
+				workload.WorkloadKindDeployment,
+				workload.WorkloadKindStatefulSet,
+			} {
+				err = errors.Join(err, r.listAndSyncWorkloadList(ctx, req, kind))
+			}
+		} else {
+			// This is a Source for a specific workload, not an entire namespace
+			err = errors.Join(err, r.syncWorkload(ctx, source))
+		}
+
+		if !v1alpha1.IsWorkloadExcludedSource(source) &&
+			!source.DeletionTimestamp.IsZero() &&
+			controllerutil.ContainsFinalizer(source, consts.DeleteInstrumentationConfigFinalizer) {
+			controllerutil.RemoveFinalizer(source, consts.DeleteInstrumentationConfigFinalizer)
+			if err := r.Update(ctx, source); err != nil {
+				return k8sutils.K8SUpdateErrorHandler(err)
+			}
+		}
+	}
+
+	return ctrl.Result{}, client.IgnoreNotFound(err)
```

Comment:
nit (already checking before):

```suggestion
	return ctrl.Result{}, nil
```

Reply:
You are referring to the top of the file, where we try to get the Source right?

This is to check here if any of the workload reconcile calls returned NotFound